src/components/Main/MoviesGrid.js

```diff
@@ -1,10 +1,34 @@
 import React from 'react';
 import MovieCard from '../movie/MovieCard';
+import {connect} from 'react-redux';
+import { fetchMovies } from '../../actions';
 
-const MoviesGrid = ({movies, onCardClick}) => (
-    <div className="movies__grid">
-        {movies.map((movie) => <MovieCard key={movie.id} movie={movie} onCardClick={onCardClick} />)}
-    </div>
-);
+const MoviesGrid = (props) => {
+    if (!props.movies.length) {
+        props.fetchMovies();
+        return null;
+    }
 
-export default MoviesGrid;
+    const movies = props.activeMovies && props.activeMovies.length ? props.activeMovies : props.movies;
+    return (
+        <div className="movies__grid">
+            {movies.map((movie) => <MovieCard key={movie.id} movie={movie}/>)}
+        </div>
+    )
+
+};
+
```

Comment:
how about this 
```
useEffect(() => {
  props.fetchMovies();
}, [])
```
thus we will trigger GET movies

and then 
```
if (!movies.length) {
   return;
}
return (<div>RENDER MOVIES HERE</div)
```

Reply:
Cool, thank you. Fixed.